### ISSUE TITLE
Questions on the code

### ISSUE THREAD

**Issue by jhb115** [rabeehk/robust-nli#2]
Hi congratulations on your paper! It is an awesome paper which I enjoyed reading!

By the way I have few questions on the code. It would be great if you could elucidate on these.

In the forward pass of BertDebiasForSequenceClassification model, what are "self.h_classifier1", "self.h_classifier1_second"?

If I understand correctly, I reckon "h_loss = loss on the hypothesis only model's prediction on the hypothesis sentence". But then, why do you take
loss = model_loss + self.lambda_h * h_loss?
(where model_loss = either CE / PoE / Focal / Rubi Loss of the main model)

According to the paper, I couldn't find an indication that the loss is defined as combination of main model's loss and hypothesis-only model's loss. So I was wondering where " + self.lambda_h * h_loss" is coming from.

And what is self.hans_only case exactly (and how does this differ to self.hypothesis_only case?) ? What's the difference between self.hans_only to self.hans case?

![Screen Shot 2021-11-15 at 1 48 12 PM](https://user-images.githubusercontent.com/40454220/141726945-92e7d183-42ca-4c35-adcc-ae6927424f5e.png)
![Screen Shot 2021-11-15 at 1 47 49 PM](https://user-images.githubusercontent.com/40454220/141726973-46ce77d6-7ac8-464c-9629-fc99f06c9a1d.png)

Thanks in advance!

**Comment by rabeehk** [comment]
Hi @jhb115 
Sorry for my late reply, please find the responses below:
- "In the forward pass of BertDebiasForSequenceClassification model, what are "self.h_classifier1", "self.h_classifier1_second"?"
from what I remember, "self.h_classifier1_second" is used when I was training an ensemble of two bias-only models, this is referring to the results reported in table 5 in paper for jointing debiasing for multiple bias patterns https://arxiv.org/pdf/1909.06321.pdf and you can find the sample to train it in https://github.com/rabeehk/robust-nli/blob/master/src/BERT/scripts/train_hans_ensemble.sh  and `h_classifier1` is used when I was training one bias-only model. 

- why do you take loss = model_loss + self.lambda_h * h_loss?
as exaplined in 3.2, we train bias-only and the base model end-to-end, then in order prevent the base model from learning the biases, the bias-only loss is not back-propagated to any shared parameters of the base model, this is done by this line: https://github.com/rabeehk/robust-nli/blob/c32ff958d4df68ac2fad9bf990f70d30eab9f297/src/BERT/utils_bert.py#L196 where we make sure the gradient is not backpropagated. 

- I have used ` self.hans` is used when I was training debiasing methods for hans dataset. Here please find the samples: https://github.com/rabeehk/robust-nli/blob/master/src/BERT/scripts/train_hans.sh 
- I was using `self.hans_only` when I was training a bias-only model for HANS dataset, this is not in an ensemble to train a robust model, only to train a bias-only model to capture the biases in HANS dataset. From what I remember, this is not used in the paper, so you can ignore this option.

thanks.
